### PR TITLE
Support interfaces endpoint

### PR DESCRIPTION
**Pull request by Andrei-Dolgolev**
Add endpoints:
### `/is_contract`
Check if address is contract on all chains wich we support.

GET `/subscriptions/is_contract?address=<contract_address>

requests example:
```
GET https://api.moonstream.to/subscriptions/is_contract?address=0x7E3353a9f992431059D7F340EF4c77016F9be8aB
Authorization: Bearer <moonstream_token>
```

response:
```
{
  "contract_info": {
    "mumbai": true
  }
}
```

If not found:
response:
404 Not Found
```
{
  "detail": "Not found contract on chains. EOA address or not used valid address."
}
```

### `/supported_interfaces`

Check if address have support interfaces.

GET `/subscriptions/supported_interfaces?address=<contract_address>&blockchain=mumbai

requests example:

ERC 1155 detected
```
GET https://api.moonstream.to/subscriptions/supported_interfaces?address=0x886D9Bb7b23BefEb22877222F4643E33f808Dd70&blockchain=mumbai
Authorization: Bearer <moonstream_token>
```

Not support erc165(interfaceID standart)
```
GET https://api.moonstream.to/subscriptions/supported_interfaces?address=0x3283300A1B1D88F161373F29591058C73C0F9167&blockchain=mumbai
Authorization: Bearer <moonstream_token>
```

ERC-20 (not detected some subtime)
```
GET https://api.moonstream.to/subscriptions/supported_interfaces?address=0x55Cd6eE720064F9c909b9B590f2c4d4aa1845416&blockchain=mumbai
Authorization: Bearer <moonstream_token>
```
response:
```
{
  "interfaces": {
    "IERC165": {
      "selector": "01ffc9a7",
      "abi": [
      ]
    }
  }
}
```

### CODE REVIEW

File: moonstreamapi/moonstreamapi/routes/subscriptions.py [hunk 1]
```diff
@@ -559,3 +566,158 @@ async def list_subscription_types() -> data.SubscriptionTypesListResponse:
...
+)
+async def address_info(request: Request, address: str):
+    """
+    Looking if address is contract
+    """
+
+    user_token = request.state.token
+
+    try:
+        Web3.toChecksumAddress(address)
```
Comment (kompotkot) [inline moonstreamapi/moonstreamapi/routes/subscriptions.py:584]: I suggest to remove here `try..catch` block and use this pattern at `ContractInfoResponse` https://github.com/moonstream-to/api/blob/d3b2e350f522879dfe6658ac0d843010b8be70c8/engineapi/engineapi/data.py#L276

File: moonstreamapi/moonstreamapi/routes/subscriptions.py [hunk 2]
```diff
@@ -559,3 +566,158 @@ async def list_subscription_types() -> data.SubscriptionTypesListResponse:
...
+    except ValueError as e:
+        raise MoonstreamHTTPException(
+            status_code=400,
+            detail=str(e),
+            internal_error=e,
+        )
+
+    try:
+        resource = bc.list_resources(
```
Comment (kompotkot) [inline moonstreamapi/moonstreamapi/routes/subscriptions.py:593]: Can be used user Bearer token for authentification

File: moonstreamapi/moonstreamapi/routes/subscriptions.py [hunk 3]
```diff
@@ -559,3 +566,158 @@ async def list_subscription_types() -> data.SubscriptionTypesListResponse:
...
+
+    contract_info = {}
+
+    for blockchain_type in AvailableBlockchainType:
+        try:
+            # connnect to blockchain
+
+            futures = []
+
+            with ThreadPoolExecutor(max_workers=5) as executor:
```
Comment (kompotkot) [inline moonstreamapi/moonstreamapi/routes/subscriptions.py:619]: I suspect we need timeouts for all thread executors, so there will be no freeze tasks

File: moonstreamapi/moonstreamapi/settings.py [hunk 4]
```diff
@@ -129,3 +156,103 @@
 BUGOUT_RESOURCE_TYPE_SUBSCRIPTION = "subscription"
 BUGOUT_RESOURCE_TYPE_ENTITY_SUBSCRIPTION = "entity_subscription"
 BUGOUT_RESOURCE_TYPE_DASHBOARD = "dashboards"
+MOONSTREAM_QUERY_TEMPLATE_CONTEXT_TYPE = "query_template"
```
[the remark this answers is not in the document]
Comment (Andrei-Dolgolev) [reply moonstreamapi/moonstreamapi/settings.py:159]:
```
entry = bc.create_entry(
            journal_id=MOONSTREAM_QUERIES_JOURNAL_ID,
            title=args.name,
            content=query,
            tags=["query_template", f"query_url:{name}"],
            context_id=name,
            context_type=MOONSTREAM_QUERY_TEMPLATE_CONTEXT_TYPE,
            context_url=name,
            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
            timeout=BUGOUT_REQUEST_TIMEOUT_SECONDS,
        )
```

File: moonstreamapi/moonstreamapi/routes/subscriptions.py (hunk 3 above)
Comment (Andrei-Dolgolev) [reply moonstreamapi/moonstreamapi/routes/subscriptions.py:618]: Add in result method.

File: moonstreamapi/moonstreamapi/settings.py [hunk 5]
```diff
@@ -110,6 +113,30 @@
...
+if MOONSTREAM_MUMBAI_WEB3_PROVIDER_URI == "":
+    raise Exception("MOONSTREAM_MUMBAI_WEB3_PROVIDER_URI env variable is not set")
+
+MOONSTREAM_XDAI_WEB3_PROVIDER_URI = os.environ.get(
+    "MOONSTREAM_XDAI_WEB3_PROVIDER_URI", ""
+)
+if MOONSTREAM_XDAI_WEB3_PROVIDER_URI == "":
+    raise Exception("MOONSTREAM_XDAI_WEB3_PROVIDER_URI env variable is not set")
+
+MOONSTREAM_WYRM_WEB3_PROVIDER_URI = os.environ.get(
```
[the remark this answers is not in the document]
Comment (Andrei-Dolgolev) [reply moonstreamapi/moonstreamapi/settings.py:134]: Add if condition.

**Review by kompotkot — APPROVED** [review]
lg

**Pull request merged**